server/libs/database/src/entities/system-config.entity.ts

```diff
@@ -5,23 +5,31 @@ export class SystemConfigEntity {
   @PrimaryColumn()
   key!: SystemConfigKey;
 
-  @Column({ type: 'varchar', nullable: true })
+  @Column({ type: 'varchar', nullable: true, transformer: { to: JSON.stringify, from: JSON.parse } })
   value!: SystemConfigValue;
 }
 
-export type SystemConfig = SystemConfigEntity[];
-
 export enum SystemConfigKey {
-  FFMPEG_CRF = 'ffmpeg_crf',
-  FFMPEG_PRESET = 'ffmpeg_preset',
-  FFMPEG_TARGET_VIDEO_CODEC = 'ffmpeg_target_video_codec',
-  FFMPEG_TARGET_AUDIO_CODEC = 'ffmpeg_target_audio_codec',
-  FFMPEG_TARGET_SCALING = 'ffmpeg_target_scaling',
+  FFMPEG = 'ffmpeg',
+  OAUTH = 'oauth',
```

Comment:
Is there a clean way in which we could build the database keys from the category and specific config interface, to get keys like `oauth_clientSecret` and `ffmpeg_targetVideoCodec`? That way we still have every config item on it's own row, which seems preferable for migrations and things like manually diving into the database (eg if you screw up your instance somehow).

Reply:
I really like the `category` and `key` columns in the db.
Preferred, these shoudl be enums/ints and not varchar as it's currently the case